### PR TITLE
[SimpleLoopUnswitch] Remove callbacks

### PR DESCRIPTION
**Pull request by boomanaiden154**
After the removal of the legacyPM version of simple loop unswitch, there is no longer a need for the callback mechanism to handle PM specific tasks. This patch removes the callbacks to help simplify the code now that they're no longer needed.

### CODE REVIEW

File: llvm/lib/Transforms/Scalar/SimpleLoopUnswitch.cpp [hunk 1]
```diff
@@ -3540,8 +3575,7 @@ unswitchLoop(Loop &L, DominatorTree &DT, LoopInfo &LI, AssumptionCache &AC,
   if (Trivial && unswitchAllTrivialConditions(L, DT, LI, SE, MSSAU)) {
     // If we unswitched successfully we will want to clean up the loop before
     // processing it further so just mark it as unswitched and return.
-    UnswitchCB(/*CurrentLoopValid*/ true, /*PartiallyInvariant*/ false,
-               /*InjectedCondition*/ false, {});
+    postUnswitch(L, LoopUpdater, L.getName(), true, false, false, {});
```
Comment (nikic) [inline llvm/lib/Transforms/Scalar/SimpleLoopUnswitch.cpp:3578]: Keep the comments on the flags here please.

**Review by nikic — APPROVED** [review]
LGTM

**Review by fhahn — APPROVED** [review]
Nice, thanks for the cleanup!

File: llvm/lib/Transforms/Scalar/SimpleLoopUnswitch.cpp [hunk 2]
```diff
@@ -3540,8 +3575,10 @@ unswitchLoop(Loop &L, DominatorTree &DT, LoopInfo &LI, AssumptionCache &AC,
   if (Trivial && unswitchAllTrivialConditions(L, DT, LI, SE, MSSAU)) {
     // If we unswitched successfully we will want to clean up the loop before
     // processing it further so just mark it as unswitched and return.
-    UnswitchCB(/*CurrentLoopValid*/ true, /*PartiallyInvariant*/ false,
-               /*InjectedCondition*/ false, {});
+    postUnswitch(/*Loop*/ L, /*LoopUpdater*/ LoopUpdater,
+                 /*LoopName*/ L.getName(),
+                 /*CurrentLoopValid*/ true, /*PartiallyInvariant*/ false,
+                 /*InjectedCondition*/ false, /*NewLoops*/ {});
```
Comment (nikic) [inline llvm/lib/Transforms/Scalar/SimpleLoopUnswitch.cpp:3581]: The comments should only be on the bool values.

**Pull request merged**